react/features/whiteboard/functions.ts

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable import/order */
+import md5 from 'js-md5';
+
+// @ts-ignore
+import { getPinnedParticipant } from '../../features/base/participants';
+import { getRemoteParticipants, isLocalParticipantModerator } from '../base/participants/functions';
+import { appendURLParam } from '../base/util/uri';
+
+// @ts-ignore
+import { getCurrentRoomId, isInBreakoutRoom } from '../breakout-rooms/functions';
+
+import { WHITEBOARD_ID } from './constants';
+
+const getWhiteboardState = (state: any): any => state['features/whiteboard'];
+
+/**
+ * Indicates whether the whiteboard is enabled in the config.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardEnabled = (state: any): boolean => state['features/base/config'].whiteboard?.enabled;
+
+/**
+ * Indicates whether the whiteboard is open.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardOpen = (state: any): boolean => getWhiteboardState(state).isOpen;
+
+/**
+ * Indicates whether the whiteboard button is visible.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardButtonVisible = (state: any): boolean =>
+    isWhiteboardEnabled(state) && (isLocalParticipantModerator(state) || isWhiteboardOpen(state));
+
+/**
+ * Indicates whether the whiteboard is present as a meeting participant.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardPresent = (state: any): boolean => getRemoteParticipants(state).has(WHITEBOARD_ID);
+
+/**
+ * Returns the whiteboard collaboration link.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {{ roomId: string, roomKey: string}|null}
+ */
+export const getCollabDetails = (state: any): {
+    roomId: string; roomKey: string;
+} => getWhiteboardState(state).collabDetails || {};
+
+/**
+ * Returns the whiteboard collaboration server url.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {string}
+ */
+export const getCollabServerUrl = (state: any): string | undefined => {
+    const collabServerBaseUrl = state['features/base/config'].whiteboard?.collabServerBaseUrl;
+
+    if (!collabServerBaseUrl) {
```

Comment:
Isn't it too late to check this here? This function should not be called if we don't have one configured.

Reply:
Hmm it shouldn't be, I'm calling this selector in the view. Even though the excalidraw instance is displayed conditionally, the whiteboard root is always present in the LargeVideo component

Reply:
Maybe the root whiteboard component can check if it's enabled / available and render null if not?